hera_cal/lst_stack/binning.py

```diff
@@ -603,6 +604,115 @@ def lst_bin_files_for_baselines(
+        - self.bin_lst: array of shape (Nlst,) containing the LST bin centres
+
+        Parameters
+        ----------
+        bl_str : str
+            The baseline string in the filenames of data to load, e.g., '0_4'.
+        configurator : LSTBinConfiguratorSingleBaseline
+            The configurator object that contains the mapping from baseline strings to data files.
+        lst_bin_edges : np.ndarray
+            Array of LST bin edges (should be one more than the number of bins).
+        lst_branch_cut : float | None
+            If provided, the LSTs will be rolled to start after this branch cut (in radians).
+        where_inpainted_file_rules : list[list[str]] | None
+            If provided, a list of pairs of strings that will be used to replace parts of the filenames to find
+            the "where inpainted" files, UVFlag files that record where that data was previously inpainted.
+        to_keep_slice : slice | None
+            For advanced users only. Typically, times are removed at the beginning and end of the data set if they have no data.
+            This option allows that behavior to be overridden with an explicit slice into an array of length len(lst_bin_edges) - 1.
+        """
+
+        self.bl_str = bl_str
+        self.configurator = configurator
+        self.lst_bin_edges = lst_bin_edges
+        self.lst_branch_cut = lst_branch_cut
+        self.where_inpainted_file_rules = where_inpainted_file_rules
+
+        # Load the data
+        self.bin_lst, self.data, self.flags, self.nsamples, self.where_inpainted, self.times_in_bins, self.lsts_in_bins = self._load_data()
+
+        # Cut out baseline dimension
+        for list_obj in (self.data, self.flags, self.nsamples, self.where_inpainted):
+            for j, arr in enumerate(list_obj):
+                # for where_inpainted, arr may be None (if where_inpainted_file_rules is None), so don't slice in that case
+                if (arr is not None) or (list_obj is not self.where_inpainted):
+                    list_obj[j] = arr[:, 0, :, :].copy()
+
+        # Roll the lists to the branch cut
+        if lst_branch_cut is not None:
+            self._roll_lists_to_lst_branch_cut()
+
+        # Figure out which times to keep and remove others
+        if to_keep_slice is None:
+            lst_has_data = np.array([~np.all(f) & (len(f) > 0) for f in self.flags])
+            ts = true_stretches(~lst_has_data)
+            start = ts[0].stop if ts[0].start == 0 else 0
+            stop = ts[-1].start if ts[-1].stop == len(self.flags) else len(self.flags)
+            self._to_keep_slice = slice(start, stop)
+        else:
+            self._to_keep_slice = to_keep_slice
+        for list_obj in self._list_objects:
+            setattr(self, list_obj, getattr(self, list_obj)[self._to_keep_slice])
+        self.bin_lst = self.bin_lst[self._to_keep_slice]
+
+    def _load_data(self):
```

Comment:
So, to get this logic in, instead of making this an instance method, make it a class method, which takes in the parameters you're currently giving to `__init__`, e.g.:

```python
@classmethod
def from_files_and_configurator(cls, 
    bl_str: str,
    configurator: LSTBinConfiguratorSingleBaseline,
    lst_bin_edges: np.ndarray,
    lst_branch_cut: float | None = None,
    where_inpainted_file_rules: list[list[str]] = None,
    to_keep_slice: slice | None = None,
): -> Self:
    # all your logic, ending with a "return cls(data, flags, ...)"
```